design/thread-manager.txt

```diff
@@ -95,6 +102,14 @@ _`.sol.thread.term.attempt`: Nonetheless, the thread manager makes a
 by moving the thread to a ring of dead threads, and avoiding scanning
 it. This might allow a malfunctioning client program to limp along.
 
+_`.sol.thread.intr`: The POSIX specification for sigaction_ says that
+if the ``SA_RESTART`` flag is set, and if "a function specified as
+interruptible is interrupted by this signal, the function shall
+restart and shall not fail with ``EINTR`` unless otherwise specified."
+
+.. |sigaction| replace:: ``sigaction()``
+.. _sigaction: https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigaction.html
+
```

Comment:
It is worth noting here (or somewhere else as well perhaps) that certain system calls do not respect the `SA_RESTART` flag. On Linux, these mostly involve timekeeping, for example `select`, `usleep`, `epoll_wait`, and pthread functions like `sem_timedwait`. This is described in `man 7 signal` on my machine, also linked here (I could not find it in pubs.opengroup.org): [man7.org](https://man7.org/linux/man-pages/man7/signal.7.html)

Reply:
I added some discussion of this issue to the "Signal and exception handling issues" section of the reference manual.